pkgs/applications/networking/jdownloader2/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv, lib, fetchurl, writeScript, makeDesktopItem, copyDesktopItems, jre, imagemagick, coreutils }:
+
+let
+
+  icon = fetchurl {
+    url = "https://jdownloader.org/_media/vote/trazo.png";
+    sha256 = "3ebab992e7dd04ffcb6c30fee1a7e2b43f3537cb2b22124b30325d25bffdac29";
+  };
+
+  wrapper = writeScript "jdownloader" ''
+    #! ${stdenv.shell}
+    PATH=${lib.makeBinPath [ jre coreutils ]}
+    JDJAR=''${XDG_DATA_HOME:-$HOME/.local/share}/jdownloader/JDownloader.jar
```

Comment:
personally, i am NOT happy with auto-updates. should be "default off, enable on demand"

Reply:
So to begin with, JDownloader supports many websites via its plugin system, and those are always autoupdated. Main app does not update automatically, only checks for updates, here is more info
https://support.jdownloader.org/Knowledgebase/Article/View/updates-update-behavior

I did not try to build from source, and I'm not sure if it is possible without access to the main repo, for example they use proprietary libraries, and I believe there are closed source parts as well (kinda like VSCode). That would require some tinkering, as for ex. AUR packages it using the bootstrapper as well.

Reply:
Here is an issue from the mirror about missing dependencies from password-protected official repo. https://github.com/mirror/jdownloader/issues/47

Reply:
> Here is an issue from the mirror about missing dependencies

the svn repos listed in https://jdownloader.org/knowledge/wiki/development/get-started are public, the issue is only a request to mirror these repos

> they use proprietary libraries, and I believe there are closed source parts as well

im afraid yes ... im trying to build from source with [headless eclipse](https://gnu-mcu-eclipse.github.io/advanced/headless-builds/), but at the end of the day, there will still be some binary blobs (`*.jar` files with closed source), so jdownloader has license `unfree`

&rarr; use containers for better security?

[pyload](https://github.com/pyload/pyload) is missing : /

```
$ nix-shell -p pyload 
error: pyload has been removed from nixpkgs, as it was unmaintained.
```

Reply:
> → use containers for better security?

https://github.com/mviereck/x11docker

Reply:
> x11docker

https://github.com/jlesage/docker-jdownloader-2 has invented this wheel already : )

Reply:
> at the end of the day, there will still be some binary blobs (`*.jar` files with closed source), so jdownloader has license `unfree`

yepp

```
Compiling 4610 source files to /build/jdownloader/bin
/build/jdownloader/src/jd/plugins/AccountInfo.java:267: error: cannot find symbol
        final long serverTime = br.getCurrentServerTime(-1);
                                  ^
  symbol:   method getCurrentServerTime(int)
  location: variable br of type Browser
```

jdownloader requires closed sources:

```
JDClosed
JDownloaderServerOptions
UpdateClientV2
JDownloaderUpdater
```

per [jdownloader/build/newBuild/build.xml](https://github.com/mirror/jdownloader/blob/master/build/newBuild/build.xml)

i want to patch the myjdownloaderclient ...

in headless mode, jdownloader requires a valid account for myjdownloader.
i have not-yet found a way to use a self-hosted api server, or to disable this completely ...
so much for "untrusted" software

relevant configkey is [serverhost](https://github.com/mirror/jdownloader/blob/1319dcb620976270cab3c32aedf54e2c3b0884dd/src/org/jdownloader/api/myjdownloader/MyJDownloaderSettings.java#L34) (default `api.jdownloader.org`) in `cfg/org.jdownloader.api.myjdownloader.MyJDownloaderSettings.json`
but jd always says

```
|---------------------------Headless Information-------------------------------
|       MyJDownloader
|       Your 'My JDownloader' logins are not correct. 
|       Please check username/email and password!
|       Enter y -> Enter Logins
|       Enter n -> Exit JDownloader
```

so im back to binary patching ... or `pyLoad`